inc/osvr/Util/Logger.h

```diff
@@ -0,0 +1,292 @@
+/** @file
+    @brief Header
+
+    @date 2016
+
+    @author
+    Sensics, Inc.
+    <http://sensics.com>
+
+*/
+
+// Copyright 2016 Sensics, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef INCLUDED_Logger_h_GUID_D8ADC0E7_A358_4FF2_960F_10F098A22F4E
+#define INCLUDED_Logger_h_GUID_D8ADC0E7_A358_4FF2_960F_10F098A22F4E
+
+// Internal Includes
+#include <osvr/Util/LogLevel.h>
+#include <osvr/Util/LineLogger.h>
+#include <osvr/Util/Export.h>
+
+// Library/third-party includes
+// - none
+
+// Standard includes
+#include <string>           // for std::string
+#include <memory>           // for std::shared_ptr
+#include <vector>           // for std::vector
+
+// Forward declarations
+
+namespace spdlog {
+class logger;
+} // end namespace spdlog
+
+namespace osvr {
+namespace util {
+namespace log {
+
+    /**
+     * @brief A wrapper around the spdlog::logger class.
+     */
+    class Logger {
+      public:
+        OSVR_UTIL_EXPORT Logger(const std::string &logger_name);
+        OSVR_UTIL_EXPORT Logger(spdlog::logger *logger);
```

Comment:
So here you can pass in a pointer - is null a valid thing to pass in, or would it be more semantically consistent to accept a reference, which can never be unseated?